### PR TITLE
feat: add VPC, Subnet, SG selection. Move from tokens to instance-profile for provider shutdown

### LINKED ISSUE #2: Feature Request: Allow using short term credentials
I'm evaluating this AWS plugin for our company, and it looks like we can only use long-term AWS credentials.

Our company moved to AWS SSO / Identity Center a while ago, and is currently using [AWS SSO CLI](https://github.com/synfinatic/aws-sso-cli) as credential process. It's not working, even with providing AWS credentials option as environment variable.

Further investigation seems to be due to the GetSessionToken call, which can only be called with long-term token.

Shouldn't the long-term token be enough without calling the API to get session token? The GetSessionToken api call only seem to be used for authenticating with MFA ([https://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html)), which we don't provide anything in this case.

### LINKED ISSUE #3: AWS_TOKEN was not removed even after expiration
This morning I attempted to pick up from where I had left off the night before using DevPod.
All commands I ran encountered the same error: `fatal get status: RequestExpired: Request has expired.`

After doing some digging, I found that the provider was attempting to use the `AWS_TOKEN` value stored in `~/.devpod/contexts/default/machines/[machine-name]/machine.json`, of which the `Expiration` value had already passed.

I removed the `AWS_TOKEN` entry from `machine.json` and was able to execute commands again.

I'm not sure what process is responsible for cleaning up this expired `AWS_TOKEN` (DevPod or the provider), but it would be a good idea for the provider to check the expiration on the token before attempting to use it.

### PR DESCRIPTION
**Pull request by 89luca89**
Some quality of life improvements:

- Add ability to specify a VPC
- Add ability to specify a Subnet for the instance
- Add ability to specify a Security Group for the instance
- Add ability to specify a Instance Profile ARN for the instance

Also:

- migrate from aws-sdk-v1 to v2 to support SSO logins

Also move from using a session token, to instance profile for shutting down.
If shutdown this way fails, it will fallback to regular shutdown.

Resolves ENG-1430
Resolves ENG-1428
Resolves ENG-1447
Resolves ENG-1432
Resolves ENG-1423
Resolves #3 
Resolved #2 
Resolves https://github.com/loft-sh/devpod/issues/313
Resolves https://github.com/loft-sh/devpod/issues/315
Resolves https://github.com/loft-sh/devpod/issues/309
Resolves https://github.com/loft-sh/devpod/issues/333

### CODE REVIEW

File: pkg/aws/aws.go [hunk 1]
```diff
@@ -149,22 +310,24 @@ func CreateDevpodSecurityGroup(provider *AwsProvider) (string, error) {
...
-					(&ec2.IpRange{}).
-						SetCidrIp("0.0.0.0/0"),
-				}),
+		IpPermissions: []types.IpPermission{
+			types.IpPermission{
+				IpProtocol: aws.String("tcp"),
+				FromPort:   aws.Int32(22),
+				ToPort:     aws.Int32(22),
+				IpRanges: []types.IpRange{
+					types.IpRange{
```
Comment (ThomasK33) [inline pkg/aws/aws.go:321]: This type is redundant here

**Review by ThomasK33 — APPROVED** [review]
LGTM

File: pkg/aws/aws.go (hunk 1 above)
Comment (89luca89) [reply pkg/aws/aws.go]: :+1:

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -149,22 +310,24 @@ func CreateDevpodSecurityGroup(provider *AwsProvider) (string, error) {
...
-			(&ec2.IpPermission{}).
-				SetIpProtocol("tcp").
-				SetFromPort(22).
-				SetToPort(22).
-				SetIpRanges([]*ec2.IpRange{
-					(&ec2.IpRange{}).
-						SetCidrIp("0.0.0.0/0"),
-				}),
+		IpPermissions: []types.IpPermission{
+			types.IpPermission{
```
[the remark this answers is not in the document]
Comment (89luca89) [reply pkg/aws/aws.go]: :+1:

**Pull request merged**